src/modules/font/freetype/TrueTypeRasterizer.cpp

```diff
@@ -236,6 +247,7 @@ FT_UInt TrueTypeRasterizer::hintingToLoadOption(Hinting hint)
 		return FT_LOAD_TARGET_LIGHT;
 	case HINTING_MONO:
 		return FT_LOAD_TARGET_MONO;
+	case HINTING_SDF:
```

Comment:
SDF freetype glyphs are still affected by whatever hinting load mode is being used (if I change the code to use different ones than `NO_HINTING` here I get different visual results).

I don't know whether it makes sense to expose SDF versus normal glyph generation as a separate option from the hinting mode, but either way making it always use `NO_HINTING` probably isn't desirable since it can stop things from looking like how the font author intended.

Reply:
Conceptually the hinting mode is a different thing from SDF versus bitmap rasterized glyphs, too.

Reply:
I assumed that because SDF rendering is made to be scalable, hinting would become irrelevant since the source and target pixel grids will almost certainly not match up.

Taking this into account, I chose to use `NO_HINTING` because it preserves the exact shape of the text, which is optimal for dynamic scaling.

Also, the resulting shape of the text (and appearance of "visual issues") depends mostly on the texture filtering settings and the original size the font was initialized at.

I understand that there are other methods more advanced than pure SDF rendering, such as MSDF which preserves sharp corners in fonts, but I think that's out of the scope of this PR.

Reply:
> because SDF rendering is made to be scalable, hinting would become irrelevant

I think it's still relevant because it combines with the input font size to affect the look of the SDF glyphs. Maybe some people might even want mono hinting with it. In this code example you can see the 'e' renders differently with normal hinting (red) versus SDF + no hinting (green), whereas SDF + normal hinting matches the non-SDF version more closely.

```Lua
font1 = love.graphics.newFont(20, "sdf")
font2 = love.graphics.newFont(20)

shader = love.graphics.newShader[[
vec4 effect(vec4 vcolor, Image tex, vec2 tc, vec2 pc)
{
	float dist = Texel(tex, tc).a;
	float edgeWidth = 1 * length(vec2(dFdx(dist), dFdy(dist)));
	float edgeDistance = 0.5;
	float opacity = smoothstep(edgeDistance - edgeWidth, edgeDistance + edgeWidth, dist);
	vcolor.a *= opacity;
	return vcolor;
}
]]

function love.draw()
	love.graphics.setShader()
	love.graphics.setColor(1, 0, 0)
	love.graphics.print("Hello World", font2, 10, 10, 0, 1)

	love.graphics.setShader(shader)
	love.graphics.setColor(0, 1, 0)
	love.graphics.print("Hello World", font1, 10, 10, 0, 1)
end
```
![Capture](https://github.com/love2d/love/assets/2662/aff7fe95-1c80-4da1-a30e-292ccb0d80c5)
![Capture2](https://github.com/love2d/love/assets/2662/1b987a37-de8f-4445-a174-32d74cb1d293)

> I think that's out of the scope of this PR.

I agree, I was just commenting that it's not a totally perfect solution to a general SDF font request (the original github issue), although it's still a nice feature to have.

Reply:
> it combines with the input font size to affect the look of the SDF glyphs. Maybe some people might even want mono hinting with it.

I hadn't even tested that before. Then instead of adding "sdf" as a HintingMode, what do you suggest? Combining it with the existing HintingModes ("sdfnormal", "sdfmono", etc.)? Adding it as another parameter to `love.graphics.newFont`? Or something else?

Reply:
I'd probably add it as a separate parameter. However the parameter list is getting kind of long so maybe it should be turned into a settings table (like what newImage etc. have) instead, since that's sort of a separate change I could do that without the sdf support part some time this week.

Reply:
> the parameter list is getting kind of long so maybe it should be turned into a settings table (like what newImage etc. have) instead.

I've done this in [5a0a6a1](https://github.com/love2d/love/commit/5a0a6a1a27dec0fbd429c98a8c4b11989fb0b719).

For the sdf setting, I think you'll just want to have a `sdf` boolean in the `Settings` struct (defaulted to false), and the table part of `luax_checktruetypesettings` would be modified to check for a `sdf` boolean field (also defaulted to false).